enhancements/network/no_overlay_mode.md

```diff
@@ -0,0 +1,965 @@
+<!--   as an intermediate step. -->
+<!-- - Minor version upgrades - you only need to support `x.N->x.N+1` upgrade -->
+<!--   steps. So, for example, it is acceptable to require a user running 4.3 to -->
+<!--   upgrade to 4.5 with a `4.3->4.4` step followed by a `4.4->4.5` step. -->
+<!-- - While an upgrade is in progress, new component versions should -->
+<!--   continue to operate correctly in concert with older component -->
+<!--   versions (aka "version skew"). For example, if a node is down, and -->
+<!--   an operator is rolling out a daemonset, the old and new daemonset -->
+<!--   pods must continue to work correctly even while the cluster remains -->
+<!--   in this partially upgraded state for some time. -->
+
+<!-- Downgrade expectations: -->
+<!-- - If an `N->N+1` upgrade fails mid-way through, or if the `N+1` cluster is -->
+<!--   misbehaving, it should be possible for the user to rollback to `N`. It is -->
+<!--   acceptable to require some documented manual steps in order to fully restore -->
+<!--   the downgraded cluster to its previous state. Examples of acceptable steps -->
+<!--   include: -->
+<!--   - Deleting any CVO-managed resources added by the new version. The -->
+<!--     CVO does not currently delete resources that no longer exist in -->
+<!--     the target version. -->
+
+There are no special concerns about upgrades, since the feature can only be turned on for the default network at installation time and for CUDNs at the time of creation of the affected CUDNs. Configuration changes should not be allowed by CNO during upgrades.
+QA coverage will include testing the upgrade of a cluster that already runs in no-overlay mode.
+
+## Version Skew Strategy
+
+### Component Version Skew
+
+During cluster upgrades, there will be version skew between ovnkube-control-plane (running on control plane nodes) and ovnkube-node (running on worker nodes). This enhancement handles version skew as follows:
+
+**Configuration Immutability:**
+- No-overlay mode configuration is immutable, preventing changes during upgrades
+- CNO will reject any attempts to modify `defaultNetworkTransport`, `defaultNetworkNoOverlayOptions`, or `bgpManagedConfig` during upgrades
+- This ensures consistent behavior across all component versions during the upgrade window
+
+**FRR Component Skew:**
+- FRR runs as a sidecar in ovnkube-node pods
+- During node rolling updates, individual nodes may have different FRR versions
+- BGP protocol ensures compatibility: established peerings remain active during rolling updates
+- Route advertisements from updated nodes should not disrupt existing routes
+
+**Testing:**
+- E2E tests will include upgrade scenarios from version N to N+1
+- Version skew between ovnkube-control-plane and ovnkube-node will be tested as part of standard rolling update procedures
+- Tests will verify that BGP peerings remain stable during upgrades
```

Comment:
Not sure I am understanding this right but FRR is not a sidecar in ovnkube-node pods?

Also, not sure I understand the focus on FRR in this enhancement.
I see a point on describing what happens when FRR is restarted or down, but not really as a version skew or upgrade thing since this enhancement does not impact FRR at all.